dev-java/commons-logging/commons-logging-1.2-r5.ebuild

```diff
@@ -55,20 +58,39 @@ src_prepare() {
 
 	if use avalon-framework; then
 		echo "avalon-framework.jar=$(java-pkg_getjars avalon-framework-4.2)" >> build.properties || die
-		EANT_GENTOO_CLASSPATH+=" avalon-framework-4.2"
 	fi
 	if use avalon-logkit; then
 		echo "logkit.jar=$(java-pkg_getjars avalon-logkit-2.0)" >> build.properties || die
-		EANT_GENTOO_CLASSPATH+=" avalon-logkit-2.0"
 	fi
 	if use log4j; then
-		echo "log4j12.jar=$(java-pkg_getjars log4j-12-api-2,log4j-core-2)" >> build.properties || die
-		EANT_GENTOO_CLASSPATH+=" log4j-12-api-2 log4j-core-2"
+		# log4j12.jar can only contain path to one single file because
+		# build.xml decides whether the Log4JLogger should be built with
+		# <available file="${log4j12.jar}" property="log4j12.present"/>,
+		# and a value that contains multiple file paths will cause the
+		# test to return a negative result.  However, classes from multiple
+		# Log4j 2 JARs are needed to compile the sources.  So, we combine
+		# them into a single JAR on the go.
+		# https://bugs.gentoo.org/834036
+		mkdir -p "${T}/log4j-2" ||
+			die "Failed to create temporary directory for Log4j 2 classes"
+		pushd "${T}/log4j-2" > /dev/null ||
+			die "Failed to enter temporary directory for Log4j 2 classes"
+		local jar="$(java-config -j)"
+		"${jar}" -xf "$(java-pkg_getjar log4j-12-api-2 log4j-12-api.jar)" ||
```

Comment:
There's nothing wrong with this at all but I wonder if for patterns like this, we should use a loop instead, e.g.

```
local target
# Could even do 12-api, api, core?
for target in log4j-12-api log4j-api log4j-core ; do
	einfo "Running ${jar} blah blah"
	"${jar}" -xf "$(java-pkg_getjar ${target}-2 ${target}.jar)" || die "Failed to extract files from ..."
done
```

Dunno, just an idea, something to keep in mind for stuff like this even if you think it's not a good idea here.

Reply:
Looks good to me, I'll try that

Reply:
Done, thanks for the suggestion!